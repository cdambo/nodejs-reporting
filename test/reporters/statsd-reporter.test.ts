import { StatsD } from "node-dogstatsd";
import StatsDReporter from "../../src/reporters/statsd-reporter";

const getReporter = (): StatsDReporter =>
  new StatsDReporter({
    host: "host",
    globalTags: { globalTag1: "global_one", globalTag2: "global_two" }
  });

describe("StatsDReporter", (): void => {
  describe("Constructor", (): void => {
    it("Creates a StatsDReporter", (): void => {
      const reporter = new StatsDReporter({
        host: "host",
        port: 5678,
        globalTags: { globalTag1: "global_one", globalTag2: "global_two" }
      });
      expect(reporter).toMatchSnapshot();
    });

    it("Creates a StatsDReporter with a default port value", (): void => {
      const reporter = new StatsDReporter({
        host: "host",
        globalTags: { globalTag1: "global_one", globalTag2: "global_two" }
      });
      expect(reporter).toMatchSnapshot();
    });
  });

  describe("timing", (): void => {
    it("Reports timing metric", (): void => {
      const mockTiming = jest.fn();
      jest.spyOn(StatsD.prototype, "timing").mockImplementation(mockTiming);
      const reporter = getReporter();
      reporter.timing("timing_test", 20, 0.5, { tag1: "one", tag2: "two" });
      expect(mockTiming.mock.calls).toMatchSnapshot();
    });
  });

  describe("increment", (): void => {
    it("Reports counter metric", (): void => {
      const mockIncrement = jest.fn();
      jest
        .spyOn(StatsD.prototype, "increment")
        .mockImplementation(mockIncrement);
      const reporter = getReporter();
      reporter.increment("increment_test", 0.5, { tag1: "one", tag2: "two" });
      expect(mockIncrement.mock.calls).toMatchSnapshot();
    });
  });

  describe("incrementBy", (): void => {
    it("Reports counter metric", (): void => {
      const mockIncrementBy = jest.fn();
      jest
        .spyOn(StatsD.prototype, "incrementBy")
        .mockImplementation(mockIncrementBy);
      const reporter = getReporter();
      reporter.incrementBy("increment_by_test", 10, {
        tag1: "one",
        tag2: "two"
      });
      expect(mockIncrementBy.mock.calls).toMatchSnapshot();
    });
  });

  describe("decrement", (): void => {
    it("Reports counter metric", (): void => {
      const mockDecrement = jest.fn();
      jest
        .spyOn(StatsD.prototype, "decrement")
        .mockImplementation(mockDecrement);
      const reporter = getReporter();
      reporter.decrement("decrement_test", 0.5, { tag1: "one", tag2: "two" });
      expect(mockDecrement.mock.calls).toMatchSnapshot();
    });
  });

  describe("decrementBy", (): void => {
    it("Reports counter metric", (): void => {
      const mockDecrementBy = jest.fn();
      jest
        .spyOn(StatsD.prototype, "decrementBy")
        .mockImplementation(mockDecrementBy);
      const reporter = getReporter();
      reporter.decrementBy("decrement_by_test", 10, {
        tag1: "one",
        tag2: "two"
      });
      expect(mockDecrementBy.mock.calls).toMatchSnapshot();
    });
  });

  describe("gauge", (): void => {
    it("Reports gauge metric", (): void => {
      const mockGauge = jest.fn();
      jest.spyOn(StatsD.prototype, "gauge").mockImplementation(mockGauge);
      const reporter = getReporter();
      reporter.gauge("gauge_test", 10, 0.5, { tag1: "one", tag2: "two" });
      expect(mockGauge.mock.calls).toMatchSnapshot();
    });
  });

  describe("histogram", (): void => {
    it("Reports histogram metric", (): void => {
      const mockHistogram = jest.fn();
      jest
        .spyOn(StatsD.prototype, "histogram")
        .mockImplementation(mockHistogram);
      const reporter = getReporter();
      reporter.histogram("histogram_test", 20, 0.5, {
        tag1: "one",
        tag2: "two"
      });
      expect(mockHistogram.mock.calls).toMatchSnapshot();
    });
  });

  describe("close", (): void => {
    it("Closes the socket", (): void => {
      const mockClose = jest.fn();
      jest.spyOn(StatsD.prototype, "close").mockImplementation(mockClose);
      const reporter = getReporter();
      reporter.close();
      expect(mockClose.mock.calls).toMatchSnapshot();
    });
  });
});
