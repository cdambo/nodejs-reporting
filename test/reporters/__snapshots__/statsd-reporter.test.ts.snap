// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StatsDReporter Constructor Creates a StatsDReporter 1`] = `
StatsDReporter {
  "client": Client {
    "ephemeral_socket": null,
    "global_tags": Array [
      "global_tag_1:global_one",
      "global_tag_2:global_two",
    ],
    "host": "host",
    "last_used_timer": null,
    "port": 5678,
    "socket": null,
  },
}
`;

exports[`StatsDReporter Constructor Creates a StatsDReporter with a default port value 1`] = `
StatsDReporter {
  "client": Client {
    "ephemeral_socket": null,
    "global_tags": Array [
      "global_tag_1:global_one",
      "global_tag_2:global_two",
    ],
    "host": "host",
    "last_used_timer": null,
    "port": 8125,
    "socket": null,
  },
}
`;

exports[`StatsDReporter close Closes the socket 1`] = `
Array [
  Array [],
]
`;

exports[`StatsDReporter decrement Reports counter metric 1`] = `
Array [
  Array [
    "decrement_test",
    0.5,
    Array [
      "tag_1:one",
      "tag_2:two",
    ],
  ],
]
`;

exports[`StatsDReporter decrementBy Reports counter metric 1`] = `
Array [
  Array [
    "decrement_by_test",
    10,
    Array [
      "tag_1:one",
      "tag_2:two",
    ],
  ],
]
`;

exports[`StatsDReporter gauge Reports gauge metric 1`] = `
Array [
  Array [
    "gauge_test",
    10,
    0.5,
    Array [
      "tag_1:one",
      "tag_2:two",
    ],
  ],
]
`;

exports[`StatsDReporter histogram Reports histogram metric 1`] = `
Array [
  Array [
    "histogram_test",
    20,
    0.5,
    Array [
      "tag_1:one",
      "tag_2:two",
    ],
  ],
]
`;

exports[`StatsDReporter increment Reports counter metric 1`] = `
Array [
  Array [
    "increment_test",
    0.5,
    Array [
      "tag_1:one",
      "tag_2:two",
    ],
  ],
]
`;

exports[`StatsDReporter incrementBy Reports counter metric 1`] = `
Array [
  Array [
    "increment_by_test",
    10,
    Array [
      "tag_1:one",
      "tag_2:two",
    ],
  ],
]
`;

exports[`StatsDReporter timing Reports timing metric 1`] = `
Array [
  Array [
    "timing_test",
    20,
    0.5,
    Array [
      "tag_1:one",
      "tag_2:two",
    ],
  ],
]
`;
